### PR TITLE
GPIB-249

### PR DESCRIPTION
**Pull request by xiangzhe0812**
To test 
1. run yarn start and navigate to http://localhost:3000/enterprise, expect see a Corporate onboarding page.
2. give all the information and submit, expect see success message show up and email send to Contact Email address

### CODE REVIEW

**Review by Mitchell139 — APPROVED** [review]

File: src/components/enterprise/EnterprisesForm.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        >
+          {message && <Alert variant="success">{message}</Alert>}
+
+          <Input label="Name" name="name" />
+          <Input label="ABN" name="abn" />
+          <Input label="Email Domain" name="emailDomain" />
+          <Input label="Contact Email" name="contactEmail" />
+          <Input label="Contact Phone Number" name="contactPhoneNumber" />
+          <Input label="Payroll Contact" name="payrollContact" />
+          <Input label="Payroll Information" name="payrollInformation" />
```
Comment (Mitchell139) [inline src/components/enterprise/EnterprisesForm.js]: The customer wants a "Number of Employees" input.

Also, it isn't clear from the names what the expected inputs for `Payroll Contact` and `Payroll Information` are.

**Review by Mitchell139 — CHANGES_REQUESTED** [review]

File: src/components/enterprise/EnterprisesForm.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      {({ isSubmitting, errors }) => (
+        <Form
+          className="enterprise-form"
+          style={{ flex: 1, width: "100%", ...style }}
+        >
+          {message && <Alert variant="success">{message}</Alert>}
+
+          <Input label="Name" name="name" />
+          <Input label="ABN" name="abn" />
+          <Input label="Email Domain" name="emailDomain" />
```
Comment (sjustintaylor) [inline src/components/enterprise/EnterprisesForm.js]: This should be more specific, or have some sort of clarification. Email domain is a little confusing. Normally for this kind of thing, we'd ask for their domain name, not their email domain.

**Comment by sjustintaylor** [comment]
A few minor changes, otherwise looks good to me

File: src/components/enterprise/validate.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { isEmail } from "validator";
+
+const validate = values => {
```
Comment (sjustintaylor) [inline src/components/enterprise/validate.js]: Why aren't we using Yup or Joi? There's no meaningful validation here. We could be properly validating the urls as well as email addresses (could also validate phone numbers based on length)

Comment (sjustintaylor) [reply src/components/enterprise/validate.js]: We could also use Yup to validate ABN numbers based on length (strip all whitespace before validating though, that way the user can add spaces)

Comment (sjustintaylor) [reply src/components/enterprise/validate.js]: As discussed, keep `validator`, add validation for urls, (validate abn on length, same for phone?)

File: src/components/enterprise/validate.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  if (!values.contactEmail) errors.contactEmail = requiredMsg;
+  if (!values.contactPhoneNumber) errors.contactPhoneNumber = requiredMsg;
+  if (!values.payrollContact) errors.payrollContact = requiredMsg;
+  if (!values.payrollInformation) errors.payrollInformation = requiredMsg;
+  if (!values.numberOfEmployers) errors.numberOfEmployers = requiredMsg;
+
+  if (!isEmail(values.contactEmail))
+    errors.contactEmail = "Please enter a valid email";
+
+  if (!isMobilePhone(values.contactPhoneNumber, "en-AU"))
```
Comment (sjustintaylor) [inline src/components/enterprise/validate.js]: This will fail on a valid landline number (Otherwise, good work on the rest of the validations)

Comment (sjustintaylor) [reply src/components/enterprise/validate.js]: If we're just asking for mobile numbers, then it needs to be specified in the form, if not then this should include something for landlines (modify this regex?: `/^(\+?61|0)4\d{8}$/`

**Review by sjustintaylor — APPROVED** [review]

**Pull request merged**